pkg/cloudprovider/providers/gce/gce_loadbalancer.go

```diff
@@ -88,11 +88,18 @@ func LoadBalancerSrcRanges() []string {
 
 // GetLoadBalancer is an implementation of LoadBalancer.GetLoadBalancer
 func (gce *GCECloud) GetLoadBalancer(clusterName string, service *v1.Service) (*v1.LoadBalancerStatus, bool, error) {
-	loadBalancerName := cloudprovider.GetLoadBalancerName(service)
+	// For backword compatibility we had to fetch the loadBalancerName this way or else a simple assignment for service.Status.LoadBalancer.Name would be enough.
+	var loadBalancerName string
+	if len(service.Status.LoadBalancer.Name) > 0 {
+		loadBalancerName = service.Status.LoadBalancer.Name
```

Comment:
As per the current design, if status name is lost, cloudprovider.DeprecatedGetLoadBalancerName(service) will be used. So, while removing cloudprovider.DeprecatedGetLoadBalancerName(service) from this snippet, we may take extra care to generate load balancer name again with gce.GetLoadBalancerName(service) if status is lost. Can't that be an alternate design? Also @thockin I didn't understand `legacy name` and `ideal name`. Are those two new variables?